src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java

```diff
@@ -150,6 +151,15 @@ public void setEnvVars(List<ContainerEnvVar> envVars) {
         this.envVars.addAll(envVars);
     }
 
+    public List<PortMapping> getPorts() {
+        return ports != null ? ports : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setPorts(List<PortMapping> ports) {
+        this.ports.addAll(ports);
```

Comment:
by this logic @carlossg the `setEnvVars` calls could also be null and fail as it just does this

```java
    public void setEnvVars(List<ContainerEnvVar> envVars) {
        this.envVars.addAll(envVars);
    }
```